cmd/cli/install_test.go

```diff
@@ -1407,57 +784,37 @@ var _ = Describe("Test envoy log level types", func() {
 	})
 })
 
-var _ = Describe("Test osm image pull policy cli option", func() {
-	It("Should correctly resolve the pull policy option to chart values", func() {
-		installCmd := &installCmd{
-			osmImagePullPolicy: "IfNotPresent",
-		}
-
-		vals, err := installCmd.resolveValues()
-		Expect(err).NotTo(HaveOccurred())
-
-		pullPolicy := vals["OpenServiceMesh"].(map[string]interface{})["image"].(map[string]interface{})["pullPolicy"]
-		Expect(pullPolicy).To(Equal("IfNotPresent"))
-	})
-
-	It("Should correctly resolve the pull policy option to chart values", func() {
-		installCmd := &installCmd{
-			osmImagePullPolicy: "Always",
-		}
-
-		vals, err := installCmd.resolveValues()
-		Expect(err).NotTo(HaveOccurred())
+func TestResolveValues(t *testing.T) {
+	assert := assert.New(t)
 
-		pullPolicy := vals["OpenServiceMesh"].(map[string]interface{})["image"].(map[string]interface{})["pullPolicy"]
-		Expect(pullPolicy).To(Equal("Always"))
-	})
-})
+	out := new(bytes.Buffer)
+	installCmd := getDefaultInstallCmd(out)
 
-var _ = Describe("Resolving values for service cert validity duration", func() {
-	It("Should correctly resolve the validity duration", func() {
-		installCmd := &installCmd{
-			serviceCertValidityDuration: "24h",
-		}
+	// Fill out fields which are empty by default
+	installCmd.containerRegistrySecret = testRegistrySecret
+	installCmd.vaultHost = testVaultHost
+	installCmd.vaultToken = testVaultToken
 
-		vals, err := installCmd.resolveValues()
-		Expect(err).NotTo(HaveOccurred())
+	expectedValues := getDefaultValues()
 
-		actual := vals["OpenServiceMesh"].(map[string]interface{})["serviceCertValidityDuration"]
-		Expect(actual).To(Equal("24h"))
-	})
-
-	It("Should correctly resolve the validity duration", func() {
-		installCmd := &installCmd{
-			serviceCertValidityDuration: "1h30m",
-		}
+	// Fill out fields which are empty by default
+	valuesConfig := []string{
+		fmt.Sprintf("OpenServiceMesh.imagePullSecrets[0].name=%s", testRegistrySecret),
+		fmt.Sprintf("OpenServiceMesh.vault.host=%s", testVaultHost),
+		fmt.Sprintf("OpenServiceMesh.vault.token=%s", testVaultToken),
+	}
+	for _, val := range valuesConfig {
+		// parses Helm strvals line and merges into a map
+		err := strvals.ParseInto(val, expectedValues)
+		assert.Nil(err)
+	}
```

Comment:
What's the purpose of this step vs. adding them to `expectedValues` by default?

Reply:
I want to keep getDefaultValues() as just what you would get from the default values for the install cmd. That way those functions can be reused in a follow up PR to replace all the lines where we copy and paste the default config.